src/components/Profile/MyEvents.vue

```diff
@@ -0,0 +1,18 @@
+<template>
+  <div>
+    <h1>My Events</h1>
+    <p>{{ username }}'s events</p>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'MyEvents',
+
+  computed: {
+    username() {
+      return this.$route.params.username;
```

Comment:
Google “route props vue router”

Reply:
I googled this, still not really sure what you're getting at. Is it related to the last comment?

Reply:
Yea kinda. So in the routes file, you want to set the myEvents route "props: true" after the component is defined. This will then allow you to take username in as a prop in this file instead of a computed route. This also means that you can make this component functional.

Reply:
https://router.vuejs.org/guide/essentials/passing-props.html#object-mode